### DISCUSSION TITLE
How does `skip_dir` work with business shared folders?

### DISCUSSION THREAD

**Discussion by hho** [abraunegg/onedrive#1643] [Q&A] [answered]
### Situation

OneDrive Business, a coworker shared a big folder with a lot of subfolders. I redacted the names, but the structure is something like this

* Team 5
  * ABC
    * testing
      * node_modules
        * …loads more folders…
    * testdata
      * dataset1
        * series123
        * series456
        * …
      * dataset2
        * logs
      * …
  * DEF
    * …
  * …

I want to exclude the `ABC` subtree (i.e. the ABC folder and everything below it) from syncing.

### What I tried

So I tried setting `skip_dir` to `/Team 5/ABC`. However, that seems to _only_ skip the ABC directory, but _not_ the folders below it. Instead, now the `testing` and `testdata` folders would be downloaded directly to my OneDrive folder instead of under `Team 5`.

Next I tried setting `skip_dir` to `/Team 5/ABC/*`. That seems to skip _exactly_ the one level below `/Team 5/ABC`, so now the `node_modules`, `dataset1` and `dataset2` would be downloaded to my OneDrive root.

All attempts were made with `onedrive v2.4.13-1~np1` installed on Ubuntu 21.04 via DEB package. I have enabled syncing business shared folders in the config and tried `onedrive --resync --dry-run --verbose` after each config change to see what would be downloaded.

### The Question

I haven't made this a bug report, because I'm not sure if this is actually a bug or if I simply don't understand the documentation. What syntax can I use to exlude an arbitrary depth of folder hierarchy?

Thanks!

**Comment by abraunegg** [comment, marked as answer]
@hho 
You are using Ubuntu ... as such, you need to make sure you critically understand https://github.com/abraunegg/onedrive/blob/master/docs/ubuntu-package-install.md#known-issues-with-installing-from-the-above-packages and *fix* the systemd service issue ....

This is potentially a bug in the 'skip_dir' pattern matching, but to achieve what you want to you should be using 'sync_list' rather than 'skip_dir' - see: https://github.com/abraunegg/onedrive/blob/master/docs/USAGE.md#selective-sync-via-sync_list-file

Please can you revisit your configuration and use 'sync_list' & open a bug for 'skip_dir' providing all the required details.
